accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -124,19 +123,23 @@ The semantics of such an enum declaration, *E*, is defined as introducing a (sem
 
   where `args` are considered as occurring in a `const` context, and it’s a **compile-time error** if they are then not compile-time constants.
 
-The resulting constructor invocations are subject to type inference, using the empty context type. *This implies that inferred type arguments to the constructor invocation itself may depend on the types of the argument expressions of `args`.* The type of the constant variable is the static type of the resulting constant object creation expression.
+  The resulting constructor invocations are subject to type inference, using the empty context type. *This implies that inferred type arguments to the constructor invocation itself may depend on the types of the argument expressions of `args`.* The type of the constant variable is the static type of the resulting constant object creation expression.
 
-  The objects created here are *not canonicalized* like other constant object creations. _(In practice, the index value is considered part of the object, so no two objects will have the same state.)_
+   The objects created here are *not canonicalized* like other constant object creations. _(In practice, the index value is considered part of the object, so no two objects will have the same state.)_
 
 - **Static `values` list**: A static constant variable named `values` is added as by the declaration  `static const List<Name> values = [id1, …, idn];`
   where `id1`…`idn` are the names of the enum entries of the `enum` declaration in source/index order.
   _If `Name` is generic, the `List<Name>` instantiates `Name` to its bounds._
 
+It's a **compile-time error** if an `enum` declaration declares or inherits (from a mixin application) a non-abstract member named  `index` which overrides the `index` getter inherited from the `Enum` class.
+
+It's a **compile-time error** if an `enum` declaration declares or inherits (from a mixin application) a non-abstract member named  `hashCode` or `==` (an `operator ==` declaration) which overrides the `hashCode` getter or `==` operator inherited from the `Object` class. (The `Enum` class does not override `hashCode` or `operator==` from `Object`). _This ensures that enum values can be used as switch statement case values, which is the main advantage of using an enum over just writing a normal class._
```

Comment:
I'm also vary about using "has".

A declaration *contains* other declarations. It's actually even a stretch to say that a declaration *inherits* anything.
The semantic class/mixin/enum that the declaration introduces may have thing or inherit things, but if there is a compile-time error, then that class/mixin/enum does not exist, so it can't "have" anything.

So, to define things in terms of concepts that are well-defined at this point in the validation, I refer only to member declarations of the member itself, or the class/interface of its superclass/superinterfaces (which are assumed well-formed before we even start looking at this declaration). The "inherits" here is really a shorthand for "the superclass has X so the enum would inherit X, if it is otherwise valid".

Reply:
Although in this case, it's probably workable, since if the enum declaration is otherwise valid, we can *then* check if it has an `index`/`hashCode`/`==` instance member which overrides the ones from `Enum`/`Object`.

I still don't like "has". It's too vague.